client/rpc-spec-v2/src/chain_head/chain_head.rs

```diff
@@ -389,12 +428,30 @@ where
 	let mut finalized_block_hashes = notification.tree_route.iter().cloned().collect::<Vec<_>>();
 	finalized_block_hashes.push(last_finalized);
 
-	let pruned_block_hashes: Vec<_> = notification.stale_heads.iter().cloned().collect();
+	// Report all pruned blocks from the notification that are not
+	// part of the `pruned_forks`.
+	let pruned_block_hashes: Vec<_> = {
+		let mut to_ignore = pruned_forks.write();
+
+		notification
+			.stale_heads
```

Comment:
It was already like this before, but `stale_heads` reports only the displaced leaves, not intermediary blocks of the displaced branch, correct?
The spec states that on finalization, all pruned blocks should be reported: https://paritytech.github.io/json-rpc-interface-spec/api/chainHead_unstable_follow.html#usage . Technically its not only the leaf of the branch that is being pruned but all blocks in between too.

Is this something that needs to be adjusted here or is this on purpose?

Reply:
Nice catch! Will fix that!